modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/EditObjectField.tsx

```diff
@@ -137,6 +157,156 @@ export default function EditObjectField({
+			value: newFilterValues,
+		};
+
+		const newObjectFieldSettings: ObjectFieldSetting[] | undefined = [
+			...(objectFieldSettings?.filter(
+				(fieldSettings) => fieldSettings.name !== 'filter'
+			) as ObjectFieldSetting[]),
+			newFilter,
+		];
+
+		const newAggregationFilters = aggregationFilters.filter(
+			(aggregationFilter) =>
+				aggregationFilter.filterBy !== objectFieldName
+		);
+
+		setAggregatonFilters(newAggregationFilters);
+
+		setValues({
+			objectFieldSettings: newObjectFieldSettings,
+		});
+	};
+
+	const handleSaveFilterColumn = (
+		filterBy?: string,
+		fieldLabel?: LocalizedValue<string>,
+		objectFieldBusinessType?: string,
+		filterType?: string,
+		objectFieldName?: string,
+		valueList?: IItem[],
+		value?: string
+	) => {
+		const newAggregationFilters = [
+			...aggregationFilters,
+			{
+				fieldLabel: fieldLabel ? fieldLabel[defaultLanguageId] : '',
+				filterBy,
+				filterType,
+				label: fieldLabel,
+				objectFieldBusinessType,
+				objectFieldName,
+				value,
+				valueList,
+			},
+		] as AggregationFilters[];
+
+		const {objectFieldSettings} = values;
+
+		const filterSetting = objectFieldSettings?.filter(
+			({name}) => name === 'filter'
+		);
+
+		if (filterSetting?.length === 0 && objectFieldSettings) {
+			const newObjectFieldSettings: ObjectFieldSetting[] | undefined = [
+				...objectFieldSettings,
+				{
+					name: 'filter',
+					value: [
+						{
+							filterBy: objectFieldName,
+							filterType,
+							value: value
+								? value
+								: valueList?.map(({value}) => value),
+						},
+					],
+				},
+			];
+
+			setAggregatonFilters(newAggregationFilters);
+			setValues({
+				objectFieldSettings: newObjectFieldSettings,
+			});
+		}
+		else {
+			if (filterSetting) {
+				const [filter] = filterSetting;
+
+				const newFilterValues: ObjectFieldFilterSetting[] = [
+					...(filter.value as ObjectFieldFilterSetting[]),
+					{
+						filterBy: objectFieldName,
+						filterType,
+						value: value
+							? value
+							: valueList?.map(({value}) => value),
+					},
+				];
+
+				const newFilter: ObjectFieldSetting = {
+					name: filter.name,
+					value: newFilterValues,
+				};
+
+				const newObjectFieldSettings:
+					| ObjectFieldSetting[]
+					| undefined = [
+					...(objectFieldSettings?.filter(
+						(fieldSetting) => fieldSetting.name !== 'filter'
+					) as ObjectFieldSetting[]),
+					newFilter,
+				];
+
+				setAggregatonFilters(newAggregationFilters);
+				setValues({
+					objectFieldSettings: newObjectFieldSettings,
+				});
+			}
+		}
+	};
+
+	useEffect(() => {
+		if (values.businessType === 'Aggregation' && objectDefinitionId2) {
+			const makeFetch = async () => {
+				const response = await fetch(
+					`/o/object-admin/v1.0/object-definitions/${objectDefinitionId2}/object-fields`,
+					{
+						headers: HEADERS,
+					}
+				);
+
+				const {
+					items: objectFields,
+				}: {
+					items: ObjectField[];
+				} = (await response.json()) as any;
```

Comment:
Hi @danvb112. Avoid using the any type. Please, check if the following change would solve it:
```suggestion
				const {
					items: objectFields,
				} = (await response.json()) as {
					items: ObjectField[];
				} ;
```